### ISSUE TITLE
Rename session parameter in primitives

### ISSUE THREAD

**Issue by jyu00** [Qiskit/qiskit-ibm-runtime#1556]
**What is the expected feature or enhancement?**

`Sampler` and `Estimator` today accept `session` and `backend` as input parameters. This is confusing to users who want to use `Batch`. We should rename `session` to `mode`, and it can take either a `Session` or a `Batch`. 

**Acceptance criteria**

- Rename `session` to `mode` with a deprecation warning

**Comment by ihincks** [comment]
Since these options are all mutually exclusive, it would seem more natural to either have them all as separate arguments (backend, session, and batch), or all the same argument (execution_mode or mode), no?


```
def __init__(self, execution_mode: IBMBackend | Session | Batch, ...):
```

**Comment by jyu00** [comment]
Yeah I agree it's better to have just a single argument. I like `mode` better only because it's shorter to type 😂
